File: fluxc/src/main/java/org/wordpress/android/fluxc/model/PostFormatModel.java
package org.wordpress.android.fluxc.model;

import com.yarolegovich.wellsql.core.Identifiable;
import com.yarolegovich.wellsql.core.annotation.Column;
import com.yarolegovich.wellsql.core.annotation.PrimaryKey;
import com.yarolegovich.wellsql.core.annotation.Table;

@Table
public class PostFormatModel implements Identifiable {
    @PrimaryKey
    @Column private int mId;

    // Site Id Foreign Key
    @Column private int mSiteId;

    // Post format attributes
    @Column private String mSlug;
    @Column private String mDisplayName;

    public int getId() {
        return mId;
    }

    public void setId(int id) {
        mId = id;
    }

    public String getDisplayName() {
        return mDisplayName;
    }

    public void setDisplayName(String displayName) {
        mDisplayName = displayName;
    }

    public String getSlug() {
        return mSlug;
    }

    public void setSlug(String slug) {
        mSlug = slug;
    }

    public int getSiteId() {
        return mSiteId;
    }

    public void setSiteId(int siteId) {
        mSiteId = siteId;
    }
}

File: fluxc/src/main/java/org/wordpress/android/fluxc/model/PostFormatModel.kt
package org.wordpress.android.fluxc.model

import com.yarolegovich.wellsql.core.Identifiable
import com.yarolegovich.wellsql.core.annotation.Column
import com.yarolegovich.wellsql.core.annotation.PrimaryKey
import com.yarolegovich.wellsql.core.annotation.Table

@Table
data class PostFormatModel(@PrimaryKey @Column private var id: Int = 0) : Identifiable {
    // Associated local site ID (could be refactored to be a FOREIGN KEY)
    @Column var siteId: Int = 0

    // Post format attributes
    @Column var slug: String? = null
    @Column var displayName: String? = null

    override fun getId(): Int = id

    override fun setId(id: Int) {
        this.id = id
    }
}
